jobs/gorouter/templates/pre-start.erb

```diff
@@ -6,10 +6,33 @@ source /var/vcap/packages/routing_utils/syslog_utils.sh
 
 tee_output_to_sys_log "${LOG_DIR}" "pre-start" <%= p("router.logging.format.timestamp") %>
 
+<%
+    ports = []
+    ports.append(p("router.port")) # has default. will always exist.
+    ports.append(p("router.status.port")) # has default. will always exist.
+    ports.append(p("router.tls_port")) # has default. will always exist.
+
+    if_p('router.prometheus.port') do |port|
+        ports.append(port)
+    end
```

Comment:
This works for some of the ports we're listening on, but what about other processes that we may or may not be in control of? e.g. a bosh-addon like bosh-dns or an antivirus daemon? 

Should we just block off anything under 10240 and make exceptions for 17001/debug ports?

Reply:
1. Some fix is better than nothing :D 
1. I would prefer to continue optimizing for the most number of connections, rather than block off everything under 10240.
1.  Here's what I have on a default cf-d
```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:8853            0.0.0.0:*               LISTEN      17708/bosh-dns-heal
tcp        0      0 127.0.0.1:2039          0.0.0.0:*               LISTEN      17814/gorouter
tcp        0      0 127.0.0.1:53080         0.0.0.0:*               LISTEN      17736/bosh-dns
tcp        0      0 127.0.0.1:2822          0.0.0.0:*               LISTEN      17759/monit-actual
tcp        0      0 127.0.0.1:2825          0.0.0.0:*               LISTEN      1310/bosh-agent
tcp        0      0 169.254.0.2:53          0.0.0.0:*               LISTEN      17736/bosh-dns
tcp        0      0 169.254.0.2:53          0.0.0.0:*               LISTEN      17736/bosh-dns
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      17814/gorouter
tcp        0      0 127.0.0.1:17002         0.0.0.0:*               LISTEN      17814/gorouter
tcp        0      0 127.0.0.1:15821         0.0.0.0:*               LISTEN      18085/discovery-reg
tcp        0      0 127.0.0.1:14823         0.0.0.0:*               LISTEN      17967/forwarder-age
tcp        0      0 127.0.0.1:14822         0.0.0.0:*               LISTEN      18002/syslog-agent
tcp        0      0 127.0.0.1:14821         0.0.0.0:*               LISTEN      18046/prom-scraper
tcp        0      0 127.0.0.1:14829         0.0.0.0:*               LISTEN      17895/udp-forwarder
tcp        0      0 127.0.0.1:14824         0.0.0.0:*               LISTEN      17931/loggregator-a
tcp        0      0 127.0.0.1:14727         0.0.0.0:*               LISTEN      18121/metrics-agent
tcp        0      0 127.0.0.1:3461          0.0.0.0:*               LISTEN      18121/metrics-agent
tcp        0      0 127.0.0.1:3460          0.0.0.0:*               LISTEN      18002/syslog-agent
tcp        0      0 127.0.0.1:3459          0.0.0.0:*               LISTEN      17931/loggregator-a
tcp        0      0 127.0.0.1:3458          0.0.0.0:*               LISTEN      17967/forwarder-age
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1021/sshd: /usr/sbi
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      17814/gorouter
tcp        0      0 0.0.0.0:14726           0.0.0.0:*               LISTEN      18121/metrics-agent
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      17814/gorouter
udp        0      0 127.0.0.1:323           0.0.0.0:*                           1637/chronyd
udp        0      0 127.0.0.1:3457          0.0.0.0:*                           17895/udp-forwarder
udp        0      0 169.254.0.2:53          0.0.0.0:*                           17736/bosh-dns
udp        0      0 169.254.0.2:53          0.0.0.0:*                           17736/bosh-dns
udp        0      0 10.0.1.9:68             0.0.0.0:*                           504/systemd-network
udp        0      0 0.0.0.0:28746           0.0.0.0:*                           17814/gorouter
```
  * I think most of the logging+metrics things choose random ephemeral ports.
  * bosh dns uses mostly udp, not tcp, so those udp ports should be safe.
  * the bosh-dns-health port is a [bosh property here](https://github.com/cloudfoundry/bosh-dns-release/blob/17020f0da21490677df966fc0367a4e845e92f9c/jobs/bosh-dns/spec#L148-L150)
  * we could hardcode that one.
1. What do you think about port `2822` and `2825` for `monit-actual` and `bosh-agent`?
1. If other things have problems, we can play whack-a-mole with them and add them to this list.

Reply:
>Is there a way to make monit start the gorouter process last, to have a greater chance of those ports being in use by the correct process when the gorouter goes to grab an next ephemeral port for a backend connection?

I don't think so. I think bosh purposefully doesn't make any promises about ordering.

Reply:
> 8853, 2822, 2825, and possibly 3458-3461 
I'll hard code these in